mbox/mbox.c

```diff
@@ -1801,11 +1801,12 @@ static enum MxStatus mbox_mbox_check_stats(struct Mailbox *m, uint8_t flags)
   {
     bool old_peek = m->peekonly;
     mx_mbox_open(m, MUTT_QUIET | MUTT_NOSORT | MUTT_PEEK);
-    mx_mbox_close(m);
-    m->peekonly = old_peek;
+    mx_mbox_close(&m);
+    if (m)
+      m->peekonly = old_peek;
   }
 
-  if (m->has_new || m->msg_new)
+  if (m && (m->has_new || m->msg_new))
```

Comment:
This is called through `mutt_mailbox_check` -> `mailbox_check` on all mailboxes -> `mx_mbox_check_stats` -> `m->mx_ops->mbox_check_stats`, which is `mbox_mbox_check_stats` for mbox. This means we'll come in with unopened mailboxes most of the times. I suggest to save the return code before calling `mx_mbox_close`.